src/main/java/org/neo4j/sql2cypher/Translator.java

```diff
@@ -461,15 +448,55 @@ else if (c instanceof QOM.IsNotNull e) {
 		}
 	}
 
-	private Node node(Table<?> t) {
+	private PatternElement resolveTableOrJoin(Table<?> t) {
+		if (t instanceof QOM.Join<?> ta && ta.$on() instanceof QOM.Eq<?> eq) {
+			var relType = eq.$arg1().getQualifiedName().last();
+			Table<?> t1 = ta.$table1();
+			Table<?> t2 = ta.$table2();
+			System.out.println("t1 = " + Arrays.toString(t1.getQualifiedName().getName()));
+			System.out.println("t2 = " + Arrays.toString(t2.getQualifiedName().getName()));
+			if (/*mightBeNodeLabel(t1.$name().first()) &&*/ resolveTableOrJoin(t1) instanceof Node from
```

Comment:
Is there any way to structually determine join tables vs. entity tables?
And how do I access the actual table name, not the alias?

Reply:
join table in my case is what represents the relationship, i.e. the many-to-many table with 2 foreign keys and a bunch of other columns (or no other columns) in many cases these don't have a primary key but are identified by the two foreign keys.

Entity tables have mostly a primary key, some data columns and any number of FK's 0..n